google/cloud/bigtable/backup.py

```diff
@@ -0,0 +1,392 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A user-friendly container for Cloud Bigtable Backup."""
+
+import re
+
+from google.cloud._helpers import _datetime_to_pb_timestamp
+from google.cloud.bigtable_admin_v2.gapic.bigtable_table_admin_client import (
+    BigtableTableAdminClient,
+)
+from google.cloud.bigtable_admin_v2.types import table_pb2
+from google.cloud.exceptions import NotFound
+from google.protobuf import field_mask_pb2
+
+_BACKUP_NAME_RE = re.compile(
+    r"^projects/(?P<project>[^/]+)/"
+    r"instances/(?P<instance_id>[a-z][-a-z0-9]*)/"
+    r"clusters/(?P<cluster_id>[a-z][-a-z0-9]*)/"
+    r"backups/(?P<backup_id>[a-z][a-z0-9_\-]*[a-z0-9])$"
+)
+
+
+class Backup(object):
+    """Representation of a Google Cloud Bigtable Backup.
+
+    A :class: `Backup` can be used to:
+
+    * :meth:`create` the backup
+    * :meth:`update` the backup
+    * :meth:`delete` the backup
+
+    :type backup_id: str
+    :param backup_id: The ID of the backup.
+
+    :type instance: :class:`~google.cloud.spanner_v1.instance.Instance`
```

Comment:
spanner?  Maybe do a quick check through code for this. My guess is some of this was lifted from the spanner impl?

Reply:
Totally. Thanks!